lib/hashie/extensions/coercion/hashie_types.rb

```diff
@@ -0,0 +1,66 @@
+module Hashie
+  module Extensions
+    module Coercion
+      module HashieTypes
+        CORE_TYPES = {
+          Integer    => :to_i,
+          Float      => :to_f,
+          Complex    => :to_c,
+          Rational   => :to_r,
+          String     => :to_s,
+          Symbol     => :to_sym
+        }
+
+        def build_coercion(type)
+          if type.is_a? Enumerable
```

Comment:
This looks like duplicated code.

I'd love it if we could figure out a way to get rid of the duplication.